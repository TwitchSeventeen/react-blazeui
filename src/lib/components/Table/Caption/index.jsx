import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Caption(props) {
  const {
    children, className, ...other
  } = props;

  const classes = classNames(
    'c-table__caption',
    className,
  );

  return (
    <caption className={classes} {...other}>
      { children }
    </caption>
  );
}

Caption.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
};

Caption.defaultProps = {
  className: null,
  children: null,
};

export default Caption;
