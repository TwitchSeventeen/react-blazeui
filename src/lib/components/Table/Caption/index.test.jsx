import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Caption from './index';

describe('Caption', () => {
  it('should be defined', () => {
    expect(Caption).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Caption>
          lorem ipsum dolor
      </Caption>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
