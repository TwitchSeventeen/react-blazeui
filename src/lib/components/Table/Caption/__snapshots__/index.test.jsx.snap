// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption renders correctly 1`] = `
<caption
  className="c-table__caption"
>
  lorem ipsum dolor
</caption>
`;
